src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControl.java

```diff
@@ -106,6 +123,17 @@
     }
 
     /**
+     * Construct a child node. The concatenation of regular expressions needs special care:
+     * see {@link #ensureSelfContainedRegex(String, boolean)} for more details.
+     * @param parent the parent node.
+     * @param subPkg the sub package name.
+     * @param regex flags interpretation of subPkg as regex pattern.
+     */
+    ImportControl(final ImportControl parent, final String subPkg, final boolean regex) {
```

Comment:
@rnveach , what do you think to create a Check to forbid "final" on arguments ?
we have a guard Check that catch reusage of argument, so this finals are just make code verbose.